client/my-sites/domains/domain-management/list/all-domains.jsx

```diff
@@ -0,0 +1,743 @@
+			this.setState( { isSavingContactInfo: false }, () =>
+				this.props.infoNotice( this.props.translate( 'No domains selected.' ) )
+			);
+			return;
+		}
+
+		if ( this.props.isContactEmailEditContext ) {
+			this.props.saveContactEmailClick();
+		}
+
+		const saveWhoisPromises = selectedDomainNamesList.map( ( domainName ) => {
+			const updatedContactInfo = this.getUpdatedContactInfo( domainName, contactInfo );
+			return wpcom.req
+				.post( `/domains/${ domainName }/whois`, {
+					whois: updatedContactInfo,
+					transfer_lock: this.state.transferLockOptOut,
+				} )
+				.then( () => {
+					this.setState( ( { contactInfoSaveResults } ) => {
+						return {
+							contactInfoSaveResults: {
+								...contactInfoSaveResults,
+								[ domainName ]: {
+									message: 'Successfully updated email address.',
+									type: 'success',
+								},
+							},
+						};
+					} );
+				} )
+				.catch( () => {
+					this.setState( ( { contactInfoSaveResults } ) => {
+						return {
+							contactInfoSaveResults: {
+								...contactInfoSaveResults,
+								[ domainName ]: {
+									message: 'Failed to update email address.',
+									type: 'error',
+								},
+							},
+						};
+					} );
+				} );
+		} );
+
+		Promise.allSettled( saveWhoisPromises ).then( () => {
+			this.setState( { isSavingContactInfo: false }, () =>
+				this.props.infoNotice( this.props.translate( 'Saving contact info is complete.' ) )
+			);
+		} );
+	};
+
+	renderActionForm() {
+		if ( ! this.isLoading() && this.props.isContactEmailEditContext ) {
+			const selectedDomainNamesList = Object.entries( this.state.selectedDomains ).reduce(
+				( result, [ domain, selected ] ) => {
+					if ( selected ) {
+						result.push( domain );
+					}
+
+					return result;
+				},
+				[]
+			);
+
+			return (
+				<Card>
+					<CardHeading>
+						{ this.props.translate( 'Edit Contact Info For Selected Domains' ) }
+					</CardHeading>
+					<BulkEditContactInfo
+						isDisabled={ this.isLoadingDomainDetails() || this.isLoadingWhoisData() }
+						isSubmitting={ this.state.isSavingContactInfo }
+						onTransferLockOptOutChange={ this.handleContactInfoTransferLockOptOutChange }
+						handleSaveContactInfo={ this.handleSaveContactInfo }
+						emailOnly={ ListAllActions.editContactEmail === this.props?.action }
+						domainNamesList={ selectedDomainNamesList }
+					/>
+				</Card>
+			);
+		}
+	}
+
+	filteredDomains() {
+		const { domainsList, canManageSitesMap } = this.props;
+		if ( ! domainsList ) {
+			return [];
+		}
+
+		// filter on sites we can manage, that aren't `wpcom` type
+		return domainsList.filter(
+			( domain ) => domain.type !== domainTypes.WPCOM && canManageSitesMap[ domain.blogId ]
+		);
+	}
+
+	renderBreadcrumbs() {
+		const { translate } = this.props;
+
+		const item = {
+			label: translate( 'All Domains' ),
+			helpBubble: translate(
+				'Manage all your domains. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Manage the domains connected to your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/all-domains-list-redesign)